torchgen/model.py

```diff
@@ -1054,6 +1205,15 @@ def __post_init__(self) -> None:
             for ret in self.returns
             if ret.annotation is not None and ret.annotation.is_write
         ]
+        immutable_returns = [ret for ret in self.returns if ret.annotation is None]
```

Comment:
Having an op that returns `(Tensor, Tensor(a))` also seems pretty weird, but I guess it's not worth banning for now (since if you really want an op that does that, you need that schema).